packages/test/stochastic-test-utils/src/types.ts

```diff
@@ -60,6 +60,7 @@ export type AsyncWeights<TOp, TState> = [
 export interface SaveInfo {
 	saveAt?: number;
 	saveOnFailure: boolean;
+	saveWhenFinish: boolean;
 	/** Filepath to dump the history file. Containing folder must already be created. */
 	filepath: string;
```

Comment:
@Abe27342 - What do you think of changing the contract for this field to not include the file extension?
This would allow the performActions.ts code to customize the full path (and automatically add a `.json` extension).
Would making such a change break a bunch of automated tooling that we already have?

Reply:
I didn't put a large amount of design thought into this feature, so no objections to changing it. That said, if we are going to do filepath manipulation from within this package, we should be exposing APIs to directly replay a fuzz test from a filepath, as otherwise the user of performFuzzActions will need to infer the same filepath manipulations to locate their file.

Making the JSON extension added by the package is arguably more OK since it's the only format that it can actually write in, but it's still a bit awkward.

Is there a problem with save info on failure being dumped to the same place as save info on success? I'm not sure this actually matters for most reasonable workflows you want to do with fuzz testing. I think I originally intended `saveAt` to be used to generate documents even on successful cases, but in practice I don't think it actually has much value except for "save after all operations". So perhaps we can replace `saveAt` with `saveOnSuccess`.

There isn't any automated tooling to deal with these replay files. And making breaking changes in the package is OK (it's @fluid-internal, thus should only be used from within this repo). But if you change the filepath logic, I'd expect you need to change other usages of `performFuzzActions` which save filepaths to disk for local debugging.

Reply:
The main drive for the filepath manipulations was that `saveAt` and `saveOnSuccess` can step on each other's output. If you think we can ditch `saveAt` and just offer `saveOnSuccess` then I'd much rather we do that.
The only other motivation is that it's nice to know from the log file whether it's the output of a run that failed, but if we really want that we can add it later by supporting multiple paths.

So, to recap, the path forward on this specific issue becomes:
* Remove the `saveAt` option
* Don't manipulate the path to add things like `_fail` or `_success`
* Don't add a `.json` extension (the caller can do that themselves)